pandas/io/common.py

```diff
@@ -132,6 +133,8 @@ def _stringify_path(filepath_or_buffer):
     if _PATHLIB_INSTALLED and isinstance(filepath_or_buffer, pathlib.Path):
         return str(filepath_or_buffer)
     if _PY_PATH_INSTALLED and isinstance(filepath_or_buffer, LocalPath):
```

Comment:
yeah I agree with @TomAugspurger here, I think we can blow aways all of these checks (and imports) and just
```
if hasattr(filepath_or_buffer, '__fspath__'):
    return str(filepath_or_buffer)
```
and things should *just* work

Reply:
I've removed the imports and checks, only the hasattr check remains.